src/main/java/com/todo/backend/Controller.java

```diff
@@ -1,54 +1,302 @@
+                }
+            }
+        } else {
+            // IF inPage EXISTS, WE TOGGLE ALL FROM THE LAST RESPONSE (IF ANY)
+            List<ToDo> aux = this.lastResponse == null ? this.DB : this.lastResponse;
+
+            int maxpage = (int) Math.max(Math.ceil(aux.size() / 10f), 1);
+            int page = (int) Math.min(Math.max(inPage, 1), maxpage);
+            List<ToDo> slice = aux.subList(10 * (page - 1), Math.min(10 * page, aux.size()));
+
+            for (ToDo toDo : slice)
+                toDo.setDone(markedAs);
+
+            found = true;
+        }
+
+        this.metrics.needsRecalculate();
+        this.lastResponse = null; // FORCE RECALCULATE
+
+        if (found)
+            return new ResponseEntity<>("Success", HttpStatus.ACCEPTED);
+        return new ResponseEntity<>("Not Found", HttpStatus.NOT_FOUND);
+    }
+
+    private List<ToDo> filter(Boolean done, String text, Priority priority) {
+        List<ToDo> filtered = new ArrayList<>(); // CLEAR ALL
+
+        for (ToDo item : DB) {
+            if (done != null && item.getDone() != done)
+                continue;
+            if (priority != null && item.getPriority() != priority)
+                continue;
+            if (text != null && !item.getText().contains(text))
+                continue;
+
+            filtered.add(item); // FILTERS PASSED
+        }
+
+        return filtered;
+    }
+
+    private void sortList(List<ToDo> filtered, Boolean sortPriority, Boolean sortDueDate) {
+        Collections.sort(filtered, new Comparator<ToDo>() {
+            @Override
+            public int compare(final ToDo item1, final ToDo item2) {
+                int diff = 0;
+                if (sortPriority != null) {
+                    if (sortPriority)
+                        diff = item1.getPriority().compareTo(item2.getPriority());
+                    else
+                        diff = item2.getPriority().compareTo(item1.getPriority());
+                }
+
+                if (sortDueDate != null && diff == 0) {
+                    Instant d1 = item1.getDue_date();
+                    Instant d2 = item2.getDue_date();
+
+                    if (d1 == null && d2 == null)
+                        return 0;
+
+                    if (sortDueDate) {
+                        diff = d1 == null ? 1
+                                : (d2 == null ? -1
+                                        : item1.getDue_date().compareTo(item2.getDue_date()));
+                    } else {
+                        diff = d1 == null ? -1
+                                : (d2 == null ? 1
+                                        : item2.getDue_date().compareTo(item1.getDue_date()));
+                    }
+                }
+
+                return diff;
+            }
+        });
+    }
+    // #endregion
+
+    // #region ################################ GET
+    @RequestMapping(method = { RequestMethod.GET }, value = { "/todos" })
+    public GETResponse getAllToDos(
```

Comment:
why not use `ResponseEntity<GETResponse>`?
also why not use a more meaningful or generic name like `TodoListWithMetrics`?